servant-contrib/src/Servant/Contrib/RegexHttpApiData.hs

```diff
@@ -0,0 +1,106 @@
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE DeriveDataTypeable  #-}
+{-# LANGUAGE KindSignatures      #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+-- | A module providing regexp validated textual data.
+--
+-- This module is intended to imported qualified
+--
+-- > import qualified Servant.Contrib.RegexHttpApiData as RE
+--
+-- The 'RE' type can be used, for example to do quick'n'dirty @servant@ apis.
+-- In the long run it's valuable to write specific types for 'Capture's or
+-- 'QueryParam's, but this type can help at the beginning.
+--
+-- @
+-- type API = "data" :> Capture "city" (RE.RE "a-zA-Z+") :> Get '[JSON] SomeData
+-- @
+module Servant.Contrib.RegexHttpApiData (
+    RE,
+    mk,
+    original,
+    groups,
+    ) where
```

Comment:
If you're not going to export everything, please use an `.Internal` module which does so.